### PR TITLE
Use a versioner to convert an internal type into an external type for beta serving

### PR DESCRIPTION
**Pull request by deads2k**
The REST storage creates internal types, not external types.  But at the time of creation, we have a scheme which includes a versioner.  We can use this, in combination with the internal object to choose the correct external type.

/kind bug
/priority important-soon
@kubernetes/sig-api-machinery-misc 
/assign @sttts 

```release-note
NONE
```

### CODE REVIEW

File: pkg/controlplane/deleted_kinds.go [hunk 1]
```diff
@@ -82,8 +85,21 @@ func newResourceExpirationEvaluator(currentVersion apimachineryversion.Info) (*r
 	return ret, nil
 }
 
-func (e *resourceExpirationEvaluator) shouldServe(resourceServingInfo rest.Storage) bool {
-	versionedPtr := resourceServingInfo.New()
+func (e *resourceExpirationEvaluator) shouldServe(gv schema.GroupVersion, versioner runtime.ObjectVersioner, resourceServingInfo rest.Storage) bool {
+	internalPtr := resourceServingInfo.New()
```
Comment (sttts) [inline pkg/controlplane/deleted_kinds.go:89]: why Ptr?

Comment (sttts) [reply pkg/controlplane/deleted_kinds.go:89]: why was this a versioned object before?

Comment (deads2k) [reply pkg/controlplane/deleted_kinds.go:89]: > why was this a versioned object before?

the name was wrong.  It was always internal .  :(

**Comment by sttts** [comment]
/lgtm

Needs rebase.

**Comment by deads2k** [comment]
/hold


the generated changes don't match my expectations

**Comment by deads2k** [comment]
I found out why the generated file changed.  The flowcontrol.apiserver.k8s.io/v1alpha1 used the prerelease tags to stop serving.  This caused the openapi to change.

**Comment by sttts** [comment]
Github and git diff is unable to see that there are only deletions in swagger.json.

Here is the `diff -u` output: https://gist.github.com/sttts/57d5dd05cc37cfcd6f73e33083a0e51e

**Comment by sttts** [comment]
These endpoints are removed from openapi:
```
-    "/apis/flowcontrol.apiserver.k8s.io/": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1alpha1/": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1alpha1/flowschemas": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1alpha1/flowschemas/{name}": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1alpha1/flowschemas/{name}/status": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1alpha1/prioritylevelconfigurations": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1alpha1/prioritylevelconfigurations/{name}": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1alpha1/prioritylevelconfigurations/{name}/status": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1alpha1/watch/flowschemas": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1alpha1/watch/flowschemas/{name}": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1alpha1/watch/prioritylevelconfigurations": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1alpha1/watch/prioritylevelconfigurations/{name}": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1beta1/": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1beta1/flowschemas": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1beta1/flowschemas/{name}": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1beta1/flowschemas/{name}/status": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1beta1/prioritylevelconfigurations": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1beta1/prioritylevelconfigurations/{name}": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1beta1/prioritylevelconfigurations/{name}/status": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1beta1/watch/flowschemas": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1beta1/watch/flowschemas/{name}": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1beta1/watch/prioritylevelconfigurations": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1beta1/watch/prioritylevelconfigurations/{name}": {
```

The v1alpha1 ones are expected, the v1beta1 ones not so much.

**Comment by deads2k** [comment]
```
[deads@localhost kubernetes]$ diff -u ../master-swagger.json api/openapi-spec/swagger.json | grep '"/apis'
-    "/apis/flowcontrol.apiserver.k8s.io/v1alpha1/": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1alpha1/flowschemas": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1alpha1/flowschemas/{name}": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1alpha1/flowschemas/{name}/status": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1alpha1/prioritylevelconfigurations": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1alpha1/prioritylevelconfigurations/{name}": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1alpha1/prioritylevelconfigurations/{name}/status": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1alpha1/watch/flowschemas": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1alpha1/watch/flowschemas/{name}": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1alpha1/watch/prioritylevelconfigurations": {
-    "/apis/flowcontrol.apiserver.k8s.io/v1alpha1/watch/prioritylevelconfigurations/{name}": {

```

**Pull request merged**